docs/src/main/sphinx/routines/case.md

```diff
@@ -0,0 +1,52 @@
+# CASE
+
+## Synopsis
+
+
+```text
+[label :] CASE condition
+  [ WHEN condition THEN statements ]
+  [ ELSE statements ]
+END CASE
+```
+
+## Description
+
+The `CASE` statement is an optional construct to allow conditional processing
+in [SQL routines](/routines/introduction).
+
+Each `condition`, after `CASE` or `WHEN` is defined as an expression. The result
+of processing the expression must result in a boolean `true` value to process
+the `statements` in the `THEN` block. After processing of the `THEN` block,
+processing jumps to the optional `ELSE` block or `END CASE`, and therefore the
+order of the `WHEN` statements is significant. A result of `false` results in
+skipping to statements in the following `WHEN` or `ELSE` blocks.
```

Comment:
Let's add a sentence at the end that says something like `Unlike other languages like C or Java, SQL does not support case fall through, so processing stops at the end of each case`... or such.  IF this isn't easy, just skip this comment

Reply:
I think this is important.. added with a bit of modification.